presentation/src/main/java/com/example/healthysalad/SelectIngredientFragment.kt

```diff
@@ -8,67 +8,132 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.Button
+import android.widget.TextView
+import androidx.core.os.bundleOf
 import androidx.recyclerview.widget.GridLayoutManager
 import androidx.recyclerview.widget.RecyclerView
 import com.example.data.Data
-
+import kotlinx.android.synthetic.main.fragment_select_ingredient.view.*
 
 /**
  * A simple [Fragment] subclass.
  * Use the [SelectIngredientFragment.newInstance] factory method to
  * create an instance of this fragment.
  */
-class SelectIngredientFragment (ingredientType : String) : Fragment() {
-    private val ingredientType = ingredientType
+
+private const val INGREDIENTTYPE  = "base"
+
+class SelectIngredientFragment () : Fragment() {
+
+    private var type: String = "base"
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
+            type = it.getString(INGREDIENTTYPE).toString()
+            Log.d("tag", type)
+        }
+    }
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
-        val view : View = inflater.inflate(R.layout.fragment_select_protein,container, false)
-        val data = Data() // make singleton class also
-        val repoData = data.repoWithKey(this.ingredientType)
+        val view : View = inflater.inflate(R.layout.fragment_select_ingredient,container, false)
+        Log.d("tag", type)
+
+        //view.select_ingredient.text = "Select " + INGREDIENTTYPE.capitalize()
+        view.select_ingredient.text = "Select " + type.capitalize()
```

Comment:
The `String.format` API or `StringBuilder` class would be more suitable for string concatenation.